### ISSUE TITLE
Quantity Discounts, Discount Type: None

### ISSUE THREAD

**Issue by chaddro** [chaddro/EasyPopulate-4.0#10]
Just discovered that Zencart will let you add quantity discounts, but allow you to set the discount type to None (0). This will, in effect, deactivate the quantity discounts. 

This presents a small issue for EP4 which will not export any quantity discounts if it finds a discount type of None. I have to check, but I believe that I also use 0 to delete existing discounts.

**Comment by mc12345678** [comment]
Confirmed that code operates as discussed above.  Consideration being given to use a fairly large status code ('999') to force the deletion, otherwise to leave the discounts as processed by ZC.  Not sure if others have "taken advantage" of this capability to suit their liking or not... May consider yet an additional switch to allow it to operate as it has, but the default to allow the continued existence.  Consideration for this to be in next revision. (4.0.31 but if not then latest of 4.0.32)

**Comment by reetp** [comment]
Are you talking about v_products_discount_type ? I use this with Pricebreaks-EP to set quantity discounts. From what I can see the 'type' I set is 2 - not sure why !

**Comment by mc12345678** [comment]
Yes, that is the field in question. The value of 2 that you use is because you are entering the actual price to be displayed rather than entering say a percentage off (1) or a "dollar" amount off (3).
 0 is used to indicate no discount, but to leave the entry available to be modified. EP4 will actually remove the entry if the value is 0. This may be unfavorable by some, but it may also be desired by others. So, yet again, instead of disrupting normal work flow at this time, I think it would be advisable to default to the ZC normal with an option to continue using EP4 as it was originally written. At some point could drop the capability, and leave the option something someone would have to "install" to make it work the way it used to... Just a thought though...

**Comment by reetp** [comment]
If it ain;t broke ;-)

Thanks for explanation. If there is a chance people use it then removing it may be a pain. I think the ability to bulk remove things is as good as the ability to bulk upload - sometimes I actually export a EP4 file, change to 'remove' and reimport it to clear out the products DB. Quick and handy.....

**Comment by mc12345678** [comment]
Overall that's the intention (to maintain the capability), but to offer it in two ways.. One is well as it currently is written which actually could upset users/new users, the other to offer a "delete" flag similar to the bulk upload to delete product...  Ideally, the method of having to enter information (taking a known action) would be the more correct way to go rather than things just "magically" disappearing...
